kratos/utilities/time_discretization.h

```diff
@@ -25,47 +26,124 @@ class ProcessInfo; // forward-declaring to not having to include it here
 
 namespace TimeDiscretization {
 
-class KRATOS_API(KRATOS_CORE) BDF1
+class KRATOS_API(KRATOS_CORE) BDF
 {
 public:
-    std::array<double, 2> ComputeBDFCoefficients(double DeltaTime) const;
-    std::array<double, 2> ComputeBDFCoefficients(const ProcessInfo& rProcessInfo) const;
+
+    KRATOS_CLASS_POINTER_DEFINITION(BDF);
+
+    /**
+     * @brief Construct a new BDF object
+     * Auxiliary constructor for derived classes
+     */
+    BDF(){};
+
+    /**
+     * @brief Construct a new BDF object
+     * Constructor with time order
+     * @param TimeOrder BDF time order
+     */
+    BDF(const unsigned int TimeOrder) : mTimeOrder(TimeOrder)
+    {
+        this->SetAuxBDFPointer(TimeOrder, mpAuxBDF);
+    };
+
+    /**
+     * @brief Destroy the BDF object
+     * Destructor of the BDF class
+     */
+    virtual ~BDF(){};
+
+    /**
+     * @brief Set the Aux BDF Pointer object
+     * This method sets a pointer to one of the auxiliary BDF
+     * classes according to the user defined time order
+     * @param TimeOrder BDF scheme time order
+     * @param rpAuxBDF Pointer to the auxiliary BDF scheme
+     */
+    static void SetAuxBDFPointer(
```

Comment:
why is this function needed?
The object was designed lightweight, and this gives the user quite some influence for manipulation

Reply:
It is needed in the constructor to select among the BDF1, BDF2, BDF3, etc. The nice point is that if you construct the "old" BDF objects the overhead is minimum. However, the new generic BDF gives more flexibility.

Reply:
Furthermore, I think we should move it to the `private` section.

Reply:
Moved to a `private` section.